ext/json/ext/generator/generator.c

```diff
@@ -789,6 +789,21 @@ struct hash_foreach_arg {
     int iter;
 };
 
+static VALUE
+convert_string_subclass(VALUE key)
+{
+    VALUE key_to_s = rb_funcall(key, i_to_s, 0);
+
+    if (RB_UNLIKELY(rb_type(key_to_s) != T_STRING)) {
```

Comment:
`RB_TYPE_P()` might be faster: https://github.com/ruby/ruby/blob/3ec7bfff2e7ac4f6f69d26676edcfd2e73ea3b05/include/ruby/internal/value_type.h#L376

Reply:
Ah it doesn't handle T_STRING explicitly currently, so I guess it's the same, but maybe in the future it might?

Reply:
This is an unlucky path, so I don't think it matter too much, but it's nicer to read, so I changed it.